### PR TITLE
feat: calculate transaction size

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: spec/ckb/types/transaction_size_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      "witnesses": [
+        "0x82df73581bcd08cb9aa270128d15e79996229ce8ea9e4f985b49fbf36762c5c37936caf3ea3784ee326f60b8992924fcf496f9503c907982525a3436f01ab32900"
+      ]
+    }
+  end
+
+  let(:transaction) { CKB::Types::Transaction.from_h(transaction_h) }
+
+  it "compute size" do
+    expect(transaction.size).to eq 536
```
Comment (shaojunda) [inline spec/ckb/types/transaction_size_spec.rb]: How is 536 calculated?

File: lib/ckb/types/transaction.rb [hunk 2]
```diff
@@ -114,6 +114,11 @@ def compute_hash
         blake2b.hexdigest
       end
 
+      def size
+        transaction_serializer = CKB::Serializers::TransactionSerializer.new(self)
+        Utils.hex_to_bin(transaction_serializer.serialize).bytesize + 4
```
Comment (shaojunda) [inline lib/ckb/types/transaction.rb]: why extra size is 4

Comment (classicalliu) [reply lib/ckb/types/transaction.rb]: I don't know.

Comment (shaojunda) [reply lib/ckb/types/transaction.rb]: Where did you get this value?

File: spec/ckb/types/transaction_size_spec.rb (hunk 1 above)
Comment (classicalliu) [reply spec/ckb/types/transaction_size_spec.rb]: https://github.com/nervosnetwork/ckb-sdk-js/pull/372/files#diff-dab93026c79d1592fe52c0e5a7f3cb6cR59
here, I add 2 Bytes data...

File: lib/ckb/types/transaction.rb (hunk 2 above)
Comment (ashchan) [reply lib/ckb/types/transaction.rb]: 4 bytes for the tx offset cost with molecule array (transactions).

Comment (classicalliu) [reply lib/ckb/types/transaction.rb]: Rename to `serialized_size_in_block`, `4` is an extra cost when add transaction to block in serialize.

File: lib/ckb/types/transaction.rb [hunk 3]
```diff
@@ -114,6 +114,11 @@ def compute_hash
         blake2b.hexdigest
       end
 
+      def serialized_size_in_block
+        transaction_serializer = CKB::Serializers::TransactionSerializer.new(self)
+        Utils.hex_to_bin(transaction_serializer.serialize).bytesize + 4
```
Comment (ashchan) [inline lib/ckb/types/transaction.rb]:
```suggestion
        serialized_tx_offset_bytesize = 4 // 4 bytes for the tx offset cost with molecule array (transactions)
        Utils.hex_to_bin(transaction_serializer.serialize).bytesize + serialized_tx_offset_bytesize
```

File: lib/ckb/types/transaction.rb [hunk 4]
```diff
@@ -114,6 +114,12 @@ def compute_hash
         blake2b.hexdigest
       end
 
+      def serialized_size_in_block
+        transaction_serializer = CKB::Serializers::TransactionSerializer.new(self)
+        serialized_tx_offset_bytesize = 4 // 4 bytes for the tx offset cost with molecule array (transactions)
```
Comment (ashchan) [inline lib/ckb/types/transaction.rb]:
```suggestion
        serialized_tx_offset_bytesize = 4 # 4 bytes for the tx offset cost with molecule array (transactions)
```

**Pull request merged**